editions/dev/tiddlers/new/Hook_ th-before-importing.tid

```diff
@@ -0,0 +1,62 @@
+created: 20210203184649726
+modified: 20210203191622629
+tags: HookMechanism
+title: Hook: th-before-importing
+type: text/vnd.tiddlywiki
+
+This hook allows plugins to inspect or modify the `importTiddler` object ''before'' any tiddlers are imported. It is invoked when the final "Import" button is clicked, but ''before'' the selected tiddlers are being imported into the store.
+
+It's part of the `NavigatorWidget.prototype.handlePerformImportEvent` function.
+
+Use this hook if you want to eg: create a customized "log-tiddler" that should contain a heading, that should only be written once.
+
+Hook function parameters:
+
+* ''importTiddler'': an object, that contains information about "selected / unselected" tiddlers and more. 
+
+Return value:
+
+* ''importTiddler'': object
+
+The original importTiddler object can be returned unmodified by the hook, which is probably the case for most use-cases.
+
+If you want to manipulate the tiddlers, that should be imported use: [[Hook: th-importing-tiddler]] instead.
```

Comment:
The comma in this sentence is unnecessary and makes it difficult to understand.

Reply:
will fix it